doc/outline.md

```diff
@@ -0,0 +1,36 @@
+documentation outline:
+
+
+- Getting started/Quick Start: 
+	
+	- Explanation of what metric learning is, and what is the purpose of this package
+	- installation
+	- a very quick example on how to import an algo (supervised or not ?) and how to do fit (and predict ?) (and split train and test) on some custom dataset (maybe sklearn.datasets.load_lfw_pairs ?)
+
+- User Guide/List of algorithms: 
+
+	- Supervised Metric Learning: (add links to examples/images from examples at the right place in the description)
+		- Problem setting
+		- Input data (+ see Preprocessor section)
+		- What you can do after fit (transform...)
+
+	- Weakly Supervised Metric Learning: (add links to examples/images from examples at the right place in the description)
+		- Problem setting
+		- Input data (+ See Preprocessor section)
+		- What you can do after fit (predict/score, tranform...)
+		- Scikit-learn compatibility (compatible with grid search + link to example of grid search)
+
+	- Usage of the Preprocessor:
+		- Purpose (performance)
+		- Use (as an argument "preprocessor" in every metric learner)
+
+
+- Examples/Tutorials: 
+	- One example with faces (prediction if same/different person) 
+	- One example of grid search to compare different algorithms (mmc, itml etc)
```

Comment:
So the goal is to do the following in order: 
1) grid search for each metric learning algorithm
2) grid search for a classifier (same classification algorithm)
3) and then compare the results using classification metrics? 

Right? Or are you only comparing the visualization after metric transformation (if so, maybe this could be combined with "Data visualisation" section)?

Reply:
I was thinking only grid search for Weakly Supervised Algorithms like MMC, the score being the accuracy score on predicting if a pair is positive/negative, since this case is the most unusual for users used to classic train/test sets of points and not pairs. But indeed we could also make a grid search on pipelines (SupervisedMetricLearner, KNN) for instance, like a regular classifier, with a classification accuracy score. 
And I don't know if you were refering to it by 1., but maybe for Supervised Learners we could also do a GridSearch with the scoring of Weakly Supervised Algorithms (we would split the points X traditionnally, but on the test set we would sample some pairs and then evaluate the accuracy of `score_pairs`) I don't know if this kind of scoring could is a traditional option for Supervised Metric Learners ? @bellet any opinion on this ? 

And I agree that it would be interesting to see the impact of hyperparameters tuning on data visualization: maybe this could make an example in the data visualization section indeed